karabo/simulation/telescope.py

```
@@ -277,27 +279,67 @@ def constructor(
+            disk_cache = FileHandler().get_tmp_dir(
+                prefix="telescope-constructor-rascil-",
+                mkdir=False,
             )
-            telescope.backend = SimulatorBackend.RASCIL
-            telescope.RASCIL_configuration = configuration
+            tm_path = os.path.join(disk_cache, f"{name}.tm")
 
+            telescope.write_to_disk(tm_path)
+            telescope.path = tm_path
             return telescope
         else:
             assert_never(backend)
 
+    @classmethod
+    def __convert_to_karabo_telescope(cls, instr_name: str) -> Telescope:
+        """Converts a site saved in RASCIl data format into a Karabo Telescope.
+
+        :param instr_name: The name of the instrument to convert.
+
+        :returns: An instance of Karabo Telescope.
+        :rtype: karabo.simulation.telescope.Telecope
+        :raises: ValueError if instr_name is not a valid RASCIL telescope
+        """
+        config: Configuration = create_named_configuration(instr_name)
+
+        site_location_gc: EarthLocation = config.location
+        # this conversion returns complex type with unit
+        # lon,lat,alt = site_location_gc.geodetic
+        longitude = site_location_gc.lon.to("deg").value
+        latitude = site_location_gc.lat.to("deg").value
+        altitude = site_location_gc.height.to("m").value
+
+        telescope = Telescope(longitude, latitude, altitude)
+        # This is used in some inteferometer simulations
+        telescope.RASCIL_configuration = config
+
+        station_coords = config.xyz.data
+        for i, coord in enumerate(station_coords):
+            telescope.add_station(
+                horizontal_x=coord[0],
+                horizontal_y=coord[1],
+                horizontal_z=coord[2],
+            )
+
+            # there are only stations in the rascil files no antennas
+            # we add a dummy antenna
+            telescope.add_antenna_to_station(i, 0.1, 0.1)
```

I don't understand why both horizontal x & y need to be 0.1. Can you elaborate?

---

Not setting a dummy antenna creates an empty file. I think that's not a goot idea. And it matches most of the telescope files in `karabo/data`. See config file `karabo/data/aca.all.tm/station000/layout.txt` for example.

---

Hmm, ok. I now had a closer look into `telescope.py` and I have some feedback/questions (not only from your code, also from previous authors):

- I see for dishes (not antenna clusters) there is a 0.1, 0.1 in other telescopes as well. However, I don't get why this is not a 0.0 instead because it's the position relative to the station center. And in my understanding, the station and dish center should be the same? Can you elaborate on this?
- To me it seems like this function only works for dish-arrays, not for antenna-arrays because of the hard-coded `telescope.add_antenna_to_station`. However, looking at possible values from `create_named_configuration` this assumption doesn't hold. I suggest to change the function that it works for both telescope-types.



NOT from your PR, but low-hanging fruit improvements:

- Please add `raise IndexError(err_msg)` to an else part in `add_antenna_to_station`. Otherwise it can cause a silent error which is not a good design choice.
- plot stations only, not also antennas in `plot_telescope_OSKAR`. They should/must have similar coordinates either way and just "spam" the plot with dots on the same location.

---

> plot stations only, not also antennas in plot_telescope_OSKAR. They should/must have similar coordinates either way and just "spam" the plot with dots on the same location.

I suggest to give it a try. For SKA-Mid each station has one antenna (dish). But there are 128 antennas for each station for SKA-Low. But the anetnnas are barely visibile on the scale of the image.